packages/components/src/notice/types.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode, SyntheticEvent } from 'react';
+
+type CommonNoticeActionProps = {
+	label: string;
+	className?: string;
+	noDefaultClasses?: boolean;
+	variant?: 'primary' | 'secondary' | 'link';
+};
+// `url` and `onClick` can both be provided, but `url` takes precedence. If
+// `url` is provided, the action's button will be rendered as an anchor and
+// `onClick` will be ignored.
+type NoticeActionWithURL = CommonNoticeActionProps & {
+	url: string;
+	onClick?: () => void;
+};
+type NoticeActionWithOnClick = CommonNoticeActionProps & {
+	url?: string;
```

Comment:
Explicitly setting these to `never` might be more helpful, as it would error if a consumer passes both.

```suggestion
type NoticeActionWithURL = CommonNoticeActionProps & {
	url: string;
	onClick?: never;
};
type NoticeActionWithOnClick = CommonNoticeActionProps & {
	url?: never;
```

(I get what you mean by "can both be provided", in the sense that it won't blow up, but I think TS is more about signaling to the consumer how it's intended to be used.)

Reply:
Ah, great idea. I like that a lot, thank you!